torch/quantization/fx/_equalize.py

```diff
@@ -449,12 +537,29 @@ def convert_eq_obs(
         elif weight_eq_obs_dict.get(node.name, None) is not None:
             weight_eq_obs = weight_eq_obs_dict.get(node.name)
             assert(isinstance(weight_eq_obs, _WeightEqualizationObserver))
-
             equalization_scale = weight_eq_obs.equalization_scale
 
-            # Scales the weights and runs the weight quantization observers
-            maybe_next_equalization_scale = maybe_get_next_equalization_scale(node, modules)
-            scale_weight_node(node, modules, equalization_scale, maybe_next_equalization_scale)
+            # Scale the weight nodes
+            if node.op == 'call_module':
+                maybe_next_equalization_scale = maybe_get_next_equalization_scale(node, modules)
+                scale_weight_node(node, modules, equalization_scale, maybe_next_equalization_scale)
+            elif node.op == 'call_function':
+                scale_weight_functional(node, model, modules, equalization_scale)
+
+                weight_eq_obs_node = maybe_get_weight_eq_obs_node(node, modules)
+                if weight_eq_obs_node is None:
+                    return
+
+                # Clear the quantization observer's min/max values so that they
+                # can get updated later based on the new scale values
+                clear_weight_quant_obs_node(node, modules)
```

Comment:
is this to restore the effect of run_weight_observers? do we still need this if we move run_weight_observers to the end?

Reply:
Yes, this is so that when we run `run_weight_observers` afterwards, the min/max observers will get updated with the scaled min/max values. I think that if we do not clear the values, then the weight quantization observers will compare the scaled min/max values against the unscaled min/max values (https://github.com/pytorch/pytorch/blob/master/torch/quantization/observer.py#L443). So for example, if the unscaled min value (`self.min_val`) is 2.0 but the scaled min value (`min_val_cur`) is 3.0, I think the resulting min value will still be 2.0.